### PR TITLE
Add rbac stuffs for mysql-exporter

### PR DESCRIPTION
**Pull request by the-redback**

### CODE REVIEW

File: pkg/controller/secret.go [hunk 1]
```diff
@@ -18,6 +18,8 @@ const (
 
 	KeyMySQLUser     = "user"
 	KeyMySQLPassword = "password"
+
+	ExporterSecretPath = "/etc/exporter/secrets"
```
Comment (tamalsaha) [inline pkg/controller/secret.go]: `/var/run/secrets/kubedb.com/`

File: pkg/controller/statefulset.go [hunk 2]
```diff
@@ -136,7 +140,29 @@ func (c *Controller) createStatefulSet(mysql *api.MySQL) (*apps.StatefulSet, kut
...
 				},
+				VolumeMounts: []core.VolumeMount{
+					{
+						Name:      "db-secret",
+						MountPath: ExporterSecretPath,
+						ReadOnly:  true,
+					},
+				},
 			})
+			in.Spec.Template.Spec.Volumes = core_util.UpsertVolume(
```
Comment (aerokite) [inline pkg/controller/statefulset.go]: Can't we pass Secret value as ENV?

File: pkg/controller/statefulset.go [hunk 3]
```diff
@@ -26,6 +27,12 @@ func (c *Controller) ensureStatefulSet(mysql *api.MySQL) (kutil.VerbType, error)
 		return kutil.VerbUnchanged, err
 	}
 
+	if c.opt.EnableRbac && mysql.GetMonitoringVendor() == mon_api.VendorPrometheus {
```
Comment (aerokite) [inline pkg/controller/statefulset.go]: What is the relation between RBAC and Monitoring?

Comment (the-redback) [reply pkg/controller/statefulset.go]: We only need rbac if monitoring exporter (side car) is needed to run. Otherwise, we don't need rbac.

Comment (aerokite) [reply pkg/controller/statefulset.go]: I don't think this is a good idea to check against Monitoring part.

File: pkg/controller/monitor.go [hunk 4]
```diff
@@ -71,11 +71,10 @@ func (c *Controller) setNewAgent(mysql *api.MySQL) error {
...
 	if mysql.Spec.Monitor != nil {
-		if oldAgent != nil {
-			if oldAgent.GetType() != mysql.Spec.Monitor.Agent {
-				if _, err := oldAgent.Delete(mysql.StatsAccessor()); err != nil {
-					log.Debugf("error in deleting Prometheus agent:", err)
-				}
+		if oldAgent != nil &&
+			oldAgent.GetType() != mysql.Spec.Monitor.Agent {
+			if _, err := oldAgent.Delete(mysql.StatsAccessor()); err != nil {
+				log.Debugf("error in deleting Prometheus agent:", err)
```
Comment (tamalsaha) [inline pkg/controller/monitor.go]: log.Errorf(

**Pull request merged**